controller/instance_manager_controller.go

```diff
@@ -327,7 +331,9 @@ func (imc *InstanceManagerController) syncInstanceManager(key string) (err error
 		}
 		switch im.Status.CurrentState {
 		case types.InstanceManagerStateRunning:
-			imc.pollProcesses()
+			if err := imc.pollProcesses(im); err != nil {
```

Comment:
This polling is not fast enough. Default resync period is 30 seconds. We can either implement a monitor like what we did with the engine controller or we can reduce the resync period to 5 seconds.

Reply:
The call to create the `SharedInformerFactory` accepts a parameter that determines the default resync period:
https://github.com/longhorn/longhorn-manager/blob/46c165bf1faac5b33bb21beb23371c597c36b8f0/controller/controller_manager.go#L60-L61
However, we clearly shouldn't change this value since it would make all of the other `SharedInformers` across all of the other `Controllers` get polled this frequently.

Based on [this issue](https://github.com/kubernetes/kubernetes/issues/26406) and [this pull request](https://github.com/kubernetes/kubernetes/pull/40759/files) on `Kubernetes` upstream, we can use the method `AddEventHandlerWithResyncPeriod` instead of the method `AddEventHandler`, which will allow us to use the same `SharedInformers` and get their benefits while still being able to resync at the rate we need.